#!/usr/bin/env python
PACKAGE = "clopema_calibration"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

"""
[ INFO] [1406109009.260942156]: DMAddShapeAlgorithm(det, "TraceBorder INSIDE BELOW 60") = 0
[ INFO] [1406109009.261052678]: DMAddShapeAlgorithm(det, "MinArea 12") = 0
[ INFO] [1406109009.261094466]: DMAddShapeAlgorithm(det, "MaxArea 400") = 0
[ INFO] [1406109009.261125119]: DMAddShapeAlgorithm(det, "TriangleNormal 3.0 0") = 0
[ INFO] [1406109009.261159231]: DMAddShapeAlgorithm(det, "MaxDerivSubpix 0.05 5 1.5 0.0") = 0
[ INFO] [1406109009.261189365]: DMAddShapeAlgorithm(det, "RequireBelief") = 0
[ INFO] [1406109009.261219812]: DMAddShapeAlgorithm(det, "MinArea 16") = 0
[ INFO] [1406109009.261251519]: DMAddShapeAlgorithm(det, "IsElliptic 0.6 1.0 0.5") = 0
[ INFO] [1406109009.261288459]: DMAddDetectionAlgorithm(det, "DotMatrix 0.2") = 0

"""

gen.add("trace_border_thr", int_t, 0, "TraceBorder Thr", 60, 0, 255)
gen.add("min_area_1", int_t, 0, "MinArea 1", 12, 0, 1000)
gen.add("max_area", int_t, 0, "MinArea", 400, 0, 1000)
gen.add("triangle_normal_minarm", double_t, 0, "TriangleNormal MinArm", 3.0, 0, 2)
gen.add("triangle_normal_minangle", double_t, 0, "TriangleNormal MinAngle", 0.0, 0, 2)
gen.add("max_deriv_subpix_res", double_t, 0, "MaxDerivSubpix Res", 0.05, 0, 10)
gen.add("max_deriv_subpix_size", double_t, 0, "MaxDerivSubpix Size", 5, 0, 10)
gen.add("max_deriv_subpix_sigma", double_t, 0, "MaxDerivSubpix Sigma", 1.5, 0, 10)
gen.add("max_deriv_subpix_corr", double_t, 0, "MaxDerivSubpix Corr", 0.0, 0, 10)
gen.add("min_area_2", int_t, 0, "MinArea 2", 16, 0, 1000)
gen.add("is_elliptic_minskew", double_t, 0, "IsElliptic MinSkew", .6, 0, 10)
gen.add("is_elliptic_maxskew", double_t, 0, "IsElliptic MaxSkew", 1.0, 0, 10)
gen.add("is_elliptic_maxdist", double_t, 0, "IsElliptic MaxDist", .5, 0, 10)
gen.add("dotmatrix_maxgriddist", double_t, 0, "Dotmatrix MaxiGridDist", .2, 0, 10)

exit(gen.generate(PACKAGE, "clopema_testbed", "Dotmatrix"))
